client/v3/client_test.go

```diff
@@ -152,23 +155,63 @@ func TestDialNoTimeout(t *testing.T) {
 }
 
 func TestIsHaltErr(t *testing.T) {
-	if !isHaltErr(context.TODO(), errors.New("etcdserver: some etcdserver error")) {
-		t.Errorf(`error prefixed with "etcdserver: " should be Halted by default`)
-	}
-	if isHaltErr(context.TODO(), rpctypes.ErrGRPCStopped) {
-		t.Errorf("error %v should not halt", rpctypes.ErrGRPCStopped)
-	}
-	if isHaltErr(context.TODO(), rpctypes.ErrGRPCNoLeader) {
-		t.Errorf("error %v should not halt", rpctypes.ErrGRPCNoLeader)
-	}
+	assert.Equal(t,
```

Comment:
I would slightly prefer to make it an 'table driven' test, as the pattern is very repetitive. 
But the current form is an already improvement. Thank you.

https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

Reply:
Please feel free to resolve this comment in a separate PR.